### PR TITLE
FIX RefererResult bug

### PR DESCRIPTION
**Pull request by valeriolopes**
When retrieving the http referer, vraptor uses the substring method to identify the beggining of the context path. If the context path appears in the URL host the return is invalid.
Example: http://application.code.com/app/test/

VRaptor will search '/app' in the URL and will find http:/**/app**lication.code.com/app.test and return /application.code.com/app/test and not /app/test as expected.

### CODE REVIEW

File: vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java [hunk 1]
```diff
@@ -150,4 +150,89 @@ public void whenRefererMatchesAControllerShouldForwardToIt() throws Exception {
...
+		when(router.parse("/anything/ok", HttpMethod.GET, request)).thenReturn(method);
+		
+		doReturn(logic).when(result).use(logic());
+		when(logic.forwardTo(RefererController.class)).thenReturn(controller);
+		
+		refererResult.forward();
+		
+		verify(logic).forwardTo(RefererController.class);
+		verify(controller).index();
+	}
```
Comment (lucascs) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Too much context to execute a simple check. Maybe it's better to relax visibility of getReferer() to protected and test only this method, what do you think?

and we get an extra extension point.

Comment (valeriolopes) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Well, that was my first commit on VRaptor so I tried to write comprehensive unit tests :-) - but I agree it's too much for something so simple. In addition it's a hard condition to achieve, the strings must match in a little-prob coincidence.

About the extension point, I don't have an opinion - would you expect this class to be extended? If so, then yes it's a good way to maintain its structure (or else people would have to create another method to perform this check, or even put its code within the callers)

Comment (lucascs) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Your tests are fine =)

it's just that this test class has several tests that could be testing only getReferer(), not the whole redirect or forward methods.

**Comment by valeriolopes** [comment]
You're right, I followed the other methods pattern but we could have simple tests which would check if the returned string matches the expected value and we're good to go. Wanna me to change that?

File: vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java (hunk 1 above)
Comment (valeriolopes) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: You're right, I followed the other methods pattern but we could have simple tests which would check if the returned string matches the expected value and we're good to go. Wanna me to change that?

Comment (Turini) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Yes, please @valeriolopes! After that change we can merge this PR.

Comment (valeriolopes) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: I ended up creating another pull request (976); I really need to stop and play with git (we still use svn here). The changes are [there](https://github.com/caelum/vraptor4/pull/976)

Comment (Turini) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Nice, thanks @valeriolopes. So I think we close this PR, right?

Comment (valeriolopes) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: Yeah, looks like the other one embraces everything

Comment (Turini) [inline vraptor-core/src/test/java/br/com/caelum/vraptor/view/DefaultRefererResultTest.java]: yeap! thks

**Pull request closed without merging**